src/ol/tilegrid.js

```diff
@@ -124,8 +127,8 @@ function resolutionsFromExtent(extent, opt_maxZoom, opt_tileSize) {
 
   const tileSize = toSize(opt_tileSize !== undefined ?
     opt_tileSize : DEFAULT_TILE_SIZE);
-  const maxResolution = Math.max(
-    width / tileSize[0], height / tileSize[1]);
+  const maxResolution = opt_maxResolution > 0 ? opt_maxResolution :
```

Comment:
```suggestion
  const maxResolution = opt_maxResolution !== undefined ? opt_maxResolution :
```

Reply:
That was intentional,  Zero (or a negative value) is not a valid resolution.

Note I have also added the option to ol/source/VectorTile as it is equally valid for sources such as that used in the **Vector tiles in EPSG:4326** example if that were styled directly via OpenLayers.

Reply:
Ok! I'm surprised that TypeScript understands it. So fine by me.